app/helpers/hosts_helper.rb

```diff
@@ -149,26 +149,9 @@ def searching?
 
   def multiple_actions
     actions = []
-    if authorized_for(:controller => :hosts, :action => :edit)
-      actions.concat [
-        [_('Change Group'), select_multiple_hostgroup_hosts_path],
-        [_('Change Environment'), select_multiple_environment_hosts_path],
-        [_('Edit Parameters'), multiple_parameters_hosts_path],
-        [_('Disable Notifications'), multiple_disable_hosts_path],
-        [_('Enable Notifications'), multiple_enable_hosts_path],
-        [_('Disassociate Hosts'), multiple_disassociate_hosts_path],
-        [_('Rebuild Config'), rebuild_config_hosts_path]
-      ]
-      actions.insert(1, [_('Build Hosts'), multiple_build_hosts_path]) if SETTINGS[:unattended]
-      actions <<  [_('Assign Organization'), select_multiple_organization_hosts_path] if SETTINGS[:organizations_enabled]
-      actions <<  [_('Assign Location'), select_multiple_location_hosts_path] if SETTINGS[:locations_enabled]
-      actions <<  [_('Change Owner'), select_multiple_owner_hosts_path]
-      actions <<  [_('Change Puppet Master'), select_multiple_puppet_proxy_hosts_path] if SmartProxy.unscoped.authorized.with_features("Puppet").exists?
-      actions <<  [_('Change Puppet CA'), select_multiple_puppet_ca_proxy_hosts_path] if SmartProxy.unscoped.authorized.with_features("Puppet CA").exists?
+    UI::HostDescription.reduce_providers(:multiple_actions).each do |provider|
+      actions += send(provider)
```

Comment:
I think you need to sort and transform into arrays before returning `actions`.

![mult-act](https://user-images.githubusercontent.com/2664090/55801032-6c1ec300-5ad5-11e9-83ea-b1d16ed08abd.png)

![select-mult](https://user-images.githubusercontent.com/2664090/55801083-8d7faf00-5ad5-11e9-87b7-be621cb2fef4.png)